pkg/crc/api/handlers.go

```diff
@@ -0,0 +1,125 @@
+package api
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+
+	"github.com/code-ready/crc/cmd/crc/cmd/config"
+	crcConfig "github.com/code-ready/crc/pkg/crc/config"
+	"github.com/code-ready/crc/pkg/crc/constants"
+	"github.com/code-ready/crc/pkg/crc/logging"
+	"github.com/code-ready/crc/pkg/crc/machine"
+	"github.com/code-ready/crc/pkg/crc/validation"
+	"github.com/code-ready/crc/pkg/crc/version"
+)
+
+func statusHandler(_ ArgsType) string {
+	statusConfig := machine.ClusterStatusConfig{Name: constants.DefaultName}
+	clusterStatus, err := machine.Status(statusConfig)
+	if err != nil {
+		logging.Error(err.Error())
+	}
+	s, err := json.Marshal(clusterStatus)
+	if err != nil {
+		return "Failed"
+	}
+	return string(s)
+}
+
+func stopHandler(_ ArgsType) string {
+	stopConfig := machine.StopConfig{
+		Name:  constants.DefaultName,
+		Debug: true,
+	}
+	commandResult, err := machine.Stop(stopConfig)
+	if err != nil {
+		logging.Error(err.Error())
+	}
+	s, err := json.Marshal(commandResult)
+	if err != nil {
+		return "Failed"
+	}
+	return string(s)
+}
+
+func startHandler(_ ArgsType) string {
+	startConfig := machine.StartConfig{
```

Comment:
This struct should be provided by the settings package. Do not populate it yourself here...

Reply:
settings package? the values are being fetched from the config properties...

Reply:
this struct should not be populated here, but from the config itself.... in this code you should only have to retrieve a config struct.

Reply:
Ahh, but this is the `startConfig` inside `machine.Start` this is again used to populate the `machineConfig struct`. This is following the same pattern as used in the start command implementation.

Reply:
OK